generators/javascript/math.js

```diff
@@ -147,60 +147,62 @@ Blockly.JavaScript['math_constant'] = function(block) {
 Blockly.JavaScript['math_number_property'] = function(block) {
   // Check if a number is even, odd, prime, whole, positive, or negative
   // or if it is divisible by certain number. Returns true or false.
-  const number_to_check = Blockly.JavaScript.valueToCode(block, 'NUMBER_TO_CHECK',
-      Blockly.JavaScript.ORDER_MODULUS) || '0';
-  const dropdown_property = block.getFieldValue('PROPERTY');
+  const PROPERTIES = {
+    'EVEN': [' % 2 == 0', Blockly.JavaScript.ORDER_MODULUS,
+        Blockly.JavaScript.ORDER_EQUALITY],
+    'ODD': [' % 2 == 1', Blockly.JavaScript.ORDER_MODULUS,
+        Blockly.JavaScript.ORDER_EQUALITY],
+    'WHOLE': [' % 1 == 0', Blockly.JavaScript.ORDER_MODULUS,
+        Blockly.JavaScript.ORDER_EQUALITY],
+    'POSITIVE': [' > 0', Blockly.JavaScript.ORDER_RELATIONAL,
+        Blockly.JavaScript.ORDER_RELATIONAL],
+    'NEGATIVE': [' < 0', Blockly.JavaScript.ORDER_RELATIONAL,
+        Blockly.JavaScript.ORDER_RELATIONAL],
+    'DIVISIBLE_BY': [null, Blockly.JavaScript.ORDER_MODULUS,
+        Blockly.JavaScript.ORDER_EQUALITY],
+    'PRIME': [null, Blockly.JavaScript.ORDER_NONE,
+        Blockly.JavaScript.ORDER_FUNCTION_CALL]
+  };
+  const dropdownProperty = block.getFieldValue('PROPERTY');
+  const tuple = PROPERTIES[dropdownProperty];
+  const suffix = tuple[0];
+  const inputOrder = tuple[1];
+  const outputOrder = tuple[2];
+  const numberToCheck = Blockly.JavaScript.valueToCode(block, 'NUMBER_TO_CHECK',
+      inputOrder) || '0';
   let code;
-  if (dropdown_property === 'PRIME') {
+  if (dropdownProperty === 'PRIME') {
     // Prime is a special case as it is not a one-liner test.
     const functionName = Blockly.JavaScript.provideFunction_(
         'mathIsPrime',
         ['function ' + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ + '(n) {',
          '  // https://en.wikipedia.org/wiki/Primality_test#Naive_methods',
-         '  if (n === 2 || n === 3) {',
+         '  if (n == 2 || n == 3) {',
```

Comment:
This should stay `==` because using `===` could result in code execution that can be different. (See Neil's example [here](https://github.com/google/blockly/issues/5678#issuecomment-961628770).)